src/internal/m365/onedrive/export.go

```diff
@@ -0,0 +1,169 @@
+package onedrive
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/m365/onedrive/metadata"
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/export"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+var _ export.Collection = &exportCollection{}
+
+// exportCollection is the implementation of export.ExportCollection for OneDrive
+type exportCollection struct {
+	// baseDir contains the path of the collection
+	baseDir string
+
+	// backingCollections will, in most cases contain just one
+	// collection. However, in cases where we have to combine multiple
+	// collections, like when generating pst files for outlook, this
+	// will contain multiple collections.
+	backingCollections []data.RestoreCollection
+
+	// version is the version of the backup this collection was part
+	// of. This is required to figure out how to get the name of the
+	// item.
+	version int
+}
+
+func (ec exportCollection) GetBasePath() string {
+	return ec.baseDir
+}
+
+func (ec exportCollection) GetItems(ctx context.Context) <-chan export.Item {
+	ch := make(chan export.Item)
+
+	go func() {
+		defer close(ch)
+
+		errs := fault.New(false)
+
+		// There will only be a single item in the backingCollections
+		// for OneDrive
+		for _, c := range ec.backingCollections {
+			for item := range c.Items(ctx, errs) {
+				itemUUID := item.UUID()
+				if isMetadataFile(itemUUID, ec.version) {
+					continue
+				}
+
+				name, err := getItemName(ctx, itemUUID, ec.version, c)
+
+				ch <- export.Item{
+					ID: itemUUID,
+					Data: export.ItemData{
+						Name: name,
+						Body: item.ToReader(),
+					},
+					Error: err,
+				}
+			}
+		}
+
+		// Return all the items that we failed to get from kopia at the end
+		for _, err := range errs.Errors().Items {
+			ch <- export.Item{
+				ID:    err.ID,
+				Error: &err,
+			}
+		}
+
+		for _, ec := range errs.Errors().Recovered {
+			ch <- export.Item{
+				// Convert recovered errors to simpler errors. These
+				// will not have an ID associated with them.
+				Error: errors.New(ec.String()),
```

Comment:
Hmm, I don't like this pattern at all.  It slices out all of the valuable structured data and traces.  Given the full handling of errors here, instead of calling `errs.Errors()`, I suggest you add a new func to fault similar to the `itemsIn` func, which iterates over the RecoveredErrors list, deduplicating and dividing the entries into items and non-items, but returning a canonical `errors` value.  That'll maximize our capacity for debugging code here.

Reply:
Ahh, yup. I should have done that from the start. I was a bit reluctant to introduce something in the `fault` package earlier for some reason.

Reply:
no worries!